src/pages/index.js

```diff
@@ -36,24 +59,103 @@ const profileEditFormPopup = new PopupWithForm({
 
 profileEditFormPopup.setEventListeners();
 
+api
+  .getUserInfo()
+  .then((userData) => {
+    profileUserInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+    })
+    profileUserInfo.setAvatarInfo({
+      link: userData.avatar,
+    })
+  })
+
 profileEditButton.addEventListener("click", function () {
   const { name, description } = profileUserInfo.getUserInfo();
   profileTitleInput.value = name;
   profileDescriptionInput.value = description;
   profileEditFormPopup.open();
 });
 
+
+/*api
+  .getInitialCards()
+  .then((cards) => {
+    cardSection.renderItems(cards);
+  })*/
+
+
 function handleProfileEditSubmit(userData) {
   const name = userData.title;
   const description = userData.description;
-  profileUserInfo.setUserInfo({ name, description });
-  profileEditFormPopup.close();
+  profileSubmitButton.textContent = "Saving...";
+  api.updateUserInfo(name, description)
+    .then((newUserData) => {
+      //newUserData includes user's id
+      profileUserInfo.setUserInfo({
+        name: newUserData.name,
+        description: newUserData.about,
+        id: newUserData._id, //capture user id
+      });
+      profileEditFormPopup.close();
+      console.log(newUserData._id);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileSubmitButton.textContent = "Save"
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```